src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -18,6 +18,35 @@ public partial class MemoryMappedFile
             HandleInheritability inheritability, MemoryMappedFileAccess access,
             MemoryMappedFileOptions options, long capacity)
         {
+            bool isSpecial = false;
+            Interop.Sys.FileStatus status = default;
+
+            if (fileStream != null)
+            {
+                int result = Interop.Sys.FStat(fileStream.SafeFileHandle, out status);
+                if (result != 0)
+                {
+                    Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
+                    throw Interop.GetExceptionForIoErrno(errorInfo);
+                }
+
+                isSpecial = (status.Mode & Interop.Sys.FileTypes.S_IFCHR) != 0;
```

Comment:
Maybe instead of naming `bool isSpecial`, name it `isRegularFile` and test for `S_IFREG` instead for the sake of clarity?

Reply:
@lpereira Are those two symbolic names actually opposites? The documentation says:

> S_IFCHR - This is the file type constant of a character-oriented device file.
> S_IFREG - This is the file type constant of a regular file.

This makes me think that is S_IFREG returns false, we can't say for sure if the file is a character-oriented device file, or a block-oriented file, or a directory, or a symbolic link, or a socket or a fifo/pipe (basically, everything that is not a regular file).

Sources:
- https://pubs.opengroup.org/onlinepubs/007908799/xsh/sysstat.h.html
- https://www.gnu.org/software/libc/manual/html_node/Testing-File-Type.html

Reply:
They're not opposites, no, but the name `isSpecial` doesn't tell me much about which kind of the file it is.  I forgot you can memory map sockets, too, so having this pivot on either "is this a regular file" or not isn't a good idea.

Maybe rename it to `isCharDev` or something of the sort, just to be explicit in kind of "special file" we're dealing with?

Reply:
my plan was to use `isRegularFile`  name (as I like it better) but keep an existing check. 
I also originally look at adding support for block devices as well, but I could not find any actual device supporting MemoryMap. That can be fixed later if we get any real use case.